### ISSUE TITLE
Use loss_weight in Model.compile() to compute total loss, but in real the total loss is more than the sum of losses

### ISSUE THREAD

**Issue by jianrui1995** [tensorflow/tensorflow#40566]
I test for the using of the loss_weight in Model.compile() ,which can weight losses. but In my test, I find the total loss is more than sum of losses.
like { 'loss': 3.4932688, 'output_1_loss': 1.107162, 'output_2_loss': 1.107162, 'output_3_loss': 1.107162}, where I made loss_weight is [1,1,1]。3.49 > 1.107*3
my version of tensorflow is 2.1.0
python is 3.6

**Comment by ravikyram** [comment]
@jianrui1995 

Request you to fill [issue template.](https://github.com/tensorflow/tensorflow/issues/new/choose).Please, share colab link or simple standalone code to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by jianrui1995** [comment]
this is my model class and main().
`class model(tf.keras.Model):
    def __init__(self,):
        super(model,self).__init__()
        self.x1 = tf.keras.layers.Dense(10,activation="relu")
        self.x2 = tf.keras.layers.Dense(10,activation="relu")
        self.pre = tf.keras.layers.Dense(3,activation="softmax")
        
    @tf.function
    def call(self, inputs, training=None, mask=None):
        out = self.x1(inputs)
        out = self.x2(out)
        out = self.pre(out)
        return [out,out,out]


def train():
    re = ra()  # a dataset.  (x,(y,y,y)) y is one-hot  in 3 dim
    mo = model()
    mo.compile(optimizer=tf.keras.optimizers.Adam(1e-3),
               loss=[MineLoss(from_logits=False),MineLoss(from_logits=False),MineLoss(from_logits=False)],
               loss_weights=[1,1,1])
    mo.fit(re().batch(4),
           epochs=5,
           validation_data=re().batch(4),
           verbose=1)

if __name__ == "__main__":
    train()`

below is my  loss class
`class MineLoss(tf.keras.losses.Loss):

def __init__(self,from_logits):

super(MineLoss,self).__init__(reduction=tf.keras.losses.Reduction.SUM_OVER_BATCH_SIZE)self.from_logits = from_logits def call(self, y_true, y_pred):

loss = tf.keras.losses.categorical_crossentropy(y_true,y_pred,from_logits=self.from_logits)return loss`

**Comment by ravikyram** [comment]
@jianrui1995 

Will it be possible to share colab link or simple standalone code with proper indentation.It will help us in localizing the issue faster.Thanks!
